frame/contracts/src/wasm/runtime.rs

```diff
@@ -1078,6 +1157,14 @@ define_env!(Env, <E: Ext>,
 	// - data_ptr - a pointer to a raw data buffer which will saved along the event.
 	// - data_len - the length of the data buffer.
 	seal_deposit_event(ctx, topics_ptr: u32, topics_len: u32, data_ptr: u32, data_len: u32) => {
+		charge_gas(ctx, RuntimeToken::DepositEvent {
+			num_topic: topics_len / sp_std::mem::size_of::<TopicOf<E::T>>() as u32,
```

Comment:
Curiously this will lead to div by zero in case of `Topic` is configured to be a ZST. I think, it's impossible for the moment cause `TopicOf` is a synonim for `T::Hash` for the moment.

Reply:
Unless you specify `#[allow(unconditional_panic)]` the compiler should catch this case. However, we should not rely on that and be more defensive. I replaced this by a `checked_div`.